### PR TITLE
add options to enable aws sdk debug log and add more logs when driver…

### LINKED ISSUE #216: Implement Deep health check in driver probe
**Is your feature request related to a problem? Please describe.**
Currently, the driver probe returns success as long as the grpc server is running. We should do deep health check in the probe to prevent issues like DNS resolution issue (eg #214)

**Describe the solution you'd like in detail**
TBD

**Additional context**
~We could also combine this with driver state reconstruction when controller starts after crash by calling into AWS to restore to the most recently know good state.~

### PR DESCRIPTION
**Pull request by AndyXiangLi**
… start up

**Is this a bug fix or adding new feature?**
Fixes #216 
**What is this PR about? / Why do we need it?**
Add a new driver level option -aws-sdk-debug-log, if set to true, aws go sdk will add debug log to the std output.
Sample for debug log
```
2021/04/15 00:41:50 DEBUG: Send Request ec2/DescribeVolumes failed, attempt 0/8, error RequestError: send request failed
caused by: Post "https://ec2.us-west-2.amazonaws.com/": dial tcp 54.240.253.45:443: i/o timeout
2021/04/15 00:41:50 DEBUG: Request ec2/DescribeVolumes Details:
```
Also adjust and update the klog level across driver.
**What testing is done?** 
unit test
e2e test

### CODE REVIEW

File: pkg/cloud/cloud_test.go [hunk 1]
```diff
@@ -67,6 +67,22 @@ func TestCreateDisk(t *testing.T) {
 			},
 			expErr: nil,
 		},
+		{
+			name:       "success: normal with gp2 options",
```
Comment (wongma7) [inline pkg/cloud/cloud_test.go:71]: is it intentional to include in same PR? It loolks fine, but, want to make sure ur not accidentally committing something wip.

File: pkg/driver/mocks/mock_cloud.go [hunk 2]
```diff
@@ -1,243 +1,508 @@
...
+	aws "github.com/aws/aws-sdk-go/aws"
+	request "github.com/aws/aws-sdk-go/aws/request"
+	ec2 "github.com/aws/aws-sdk-go/service/ec2"
 	gomock "github.com/golang/mock/gomock"
 	cloud "github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/cloud"
+	reflect "reflect"
 )
 
-// MockCloud is a mock of Cloud interface.
+// MockEC2 is a mock of EC2 interface
```
Comment (wongma7) [inline pkg/driver/mocks/mock_cloud.go:17]: how come it's mocking ec2 interface now? the ec2 mock should go to mock_ec2.go (looking at script https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/hack/update-gomock#L21)

Comment (AndyXiangLi) [reply pkg/driver/mocks/mock_cloud.go:17]: Interesting.. Can you try on your end? I just ran make bin/mockgen under project root path and get this EC2 mock added into the mock_cloud.go

File: pkg/cloud/cloud_test.go (hunk 1 above)
Comment (AndyXiangLi) [reply pkg/cloud/cloud_test.go:71]: Yeah.. Added this to improve some coverage on cloud.go we can remove this if you want to put it separately..

Comment (wongma7) [reply pkg/cloud/cloud_test.go:71]: fine to keep it here :+1:

File: pkg/driver/mocks/mock_cloud.go (hunk 2 above)
Comment (wongma7) [reply pkg/driver/mocks/mock_cloud.go:17]: I tried with latest mockgen, it only changed my go.sum : /
```
~/g/s/g/k/aws-ebs-csi-driver (master|✔) $ rm bin/mockgen 
~/g/s/g/k/aws-ebs-csi-driver (master|✔) $ make mockgen
go get github.com/golang/mock/mockgen@latest
go: found github.com/golang/mock/mockgen in github.com/golang/mock v1.5.0
go: downloading golang.org/x/tools v0.0.0-20191119224855-298f0cb1881e
go: downloading golang.org/x/mod v0.3.0
./hack/update-gomock
```

Comment (AndyXiangLi) [reply pkg/driver/mocks/mock_cloud.go]: figured that out.. I didn't specify the interface name when I ran the command.. updated now

**Comment by AndyXiangLi** [comment]
@ayberk migration test is consistently failing the first time, any thoughts?

**Pull request merged**